### PR TITLE
add gd32f470 support

### PR DESCRIPTION
**Pull request by GD32-MCU**

### CODE REVIEW

File: boards/arm/gd32f4/gd32f470zk-eval/Kconfig [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+			TX    USART0_TX PA9
+			-- ----- --------- ---
+
+config GD32F470ZK_EVAL_CONSOLE_BOARD
+	bool "Eval Com Connector"
+	select GD32_USART0
+	select USART0_SERIALDRIVER
+
+config GD32F470ZK_EVAL_CONSOLE_VIRTUAL
+	bool "Virtual Comport"
```
Comment (acassis) [inline boards/arm/gd32f4/gd32f470zk-eval/Kconfig:41]:
```suggestion
	bool "Virtual ComPort"

File: boards/arm/gd32f4/gd32f470zk-eval/Kconfig [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	select GD32_USART3
+	select USART3_SERIALDRIVER
+
+config GD32F470ZK_EVAL_CONSOLE_NONE
+	bool "No Console"
```
[the remark this answers is not in the document]
Comment (acassis) [reply boards/arm/gd32f4/gd32f470zk-eval/Kconfig:46]: User still can use Console over USB or LWL

**Comment by acassis** [comment]
@xiaoxiang781216 why sometimes we get this error and sometimes we don't ?
```
====================================================================================
Configuration/Tool: c5471evm/httpd,CONFIG_ARM_TOOLCHAIN_GNU_EABI
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
/usr/bin/bash: arm-nuttx-eabi-gcc: command not found
/usr/bin/bash: arm-nuttx-eabi-gcc: command not found
/usr/bin/bash: arm-nuttx-eabi-gcc: command not found
/usr/bin/bash: arm-nuttx-eabi-gcc: command not found
/usr/bin/bash: arm-nuttx-eabi-gcc: command not found
/usr/bin/bash: arm-nuttx-eabi-gcc: command not found
/usr/bin/bash: arm-nuttx-eabi-gcc: command not found
/usr/bin/bash: arm-nuttx-eabi-gcc: command not found
/usr/bin/bash: arm-nuttx-eabi-gcc: command not found
/usr/bin/bash: arm-nuttx-eabi-gcc: command not found
```

**Comment by pkarashchenko** [comment]
Seems like some defconfigs should be normalised like `boards/arm/gd32f4/gd32f470zk-eval/configs/spilfsnsh/defconfig`

**Comment by xiaoxiang781216** [comment]
> Seems like some defconfigs should be normalised like `boards/arm/gd32f4/gd32f470zk-eval/configs/spilfsnsh/defconfig`

Yes, @GD32-MCU please run and amend the change:
```
./tools/refresh.sh --silent gd32f470zk-eval/spilfsnsh
```

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 why sometimes we get this error and sometimes we don't ?
> 
> ```
> ====================================================================================
> Configuration/Tool: c5471evm/httpd,CONFIG_ARM_TOOLCHAIN_GNU_EABI
> ------------------------------------------------------------------------------------
>   Cleaning...
>   Configuring...
> /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> ```

@acassis Because c5471evm/httpd select arm-nuttx-eabi-gcc in defconfig, but the docker used by ci doesn't have it.
Here is the basic flow:

1. ./configure.sh c5471evm/httpd
2. replace arm-nuttx-eabi-gcc with arm-none-eabi-gcc by tweaking .config
3. make -jxx

The warning happens in the first step.

**Comment by acassis** [comment]
> > @xiaoxiang781216 why sometimes we get this error and sometimes we don't ?
> > ```
> > ====================================================================================
> > Configuration/Tool: c5471evm/httpd,CONFIG_ARM_TOOLCHAIN_GNU_EABI
> > ------------------------------------------------------------------------------------
> >   Cleaning...
> >   Configuring...
> > /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> > /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> > /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> > /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> > /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> > /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> > /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> > /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> > /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> > /usr/bin/bash: arm-nuttx-eabi-gcc: command not found
> > ```
> 
> @acassis Because c5471evm/httpd select arm-nuttx-eabi-gcc in defconfig, but the docker used by ci doesn't have it. Here is the basic flow:
> 
>     1. ./configure.sh c5471evm/httpd
> 
>     2. replace arm-nuttx-eabi-gcc with arm-none-eabi-gcc by tweaking .config
> 
>     3. make -jxx
> 
> 
> The warning happens in the first step could be ignored safely.

@xiaoxiang781216 thank you, so this message are just warnings, right?

**Comment by xiaoxiang781216** [comment]
Yes.

File: boards/arm/gd32f4/gd32f470zk-eval/Kconfig (hunk 2 above)
[the remark this answers is not in the document]
Comment (GD32-MCU) [reply boards/arm/gd32f4/gd32f470zk-eval/Kconfig:46]: Thanks for the adcvice, I, change it

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**